usaspending_api/etl/management/commands/swap_in_new_table.py

```diff
@@ -0,0 +1,292 @@
+import logging
+import re
+
+from django.core.management import BaseCommand
+from django.db import connection, transaction
+
+from usaspending_api.common.helpers.sql_helpers import ordered_dictionary_fetcher
+
+logger = logging.getLogger("script")
+
+
+class Command(BaseCommand):
+    help = """
+    This command is used to swap two tables; the current and a new table with "_temp" appended.
+    Validation is run against the new table to ensure that after the swap is complete all of the indexes, constraints,
+    columns, and table name will be the same.
+
+    NOTE: This entire process IS NOT ATOMIC (only the final swap of tables)!
+    This choice was made to prevent potential deadlock scenarios since the swapping / renaming of indexes, constraints,
+    and tables should only take milliseconds. If this command fails then any cleanup will have to be done manually.
+    Any failure in this command should not affect current API since the final "DROP TABLE" command is not run
+    until the very end.
```

Comment:
Is it worth noting or trying to figure out what to note about how queries running against the dropped table will be affected?

I think they are allowed to complete so long as a vacuum or autovacuum doesn't remove the table's tuples (which would be marked inactive after the drop, but may still be in the MVCC system). That's my understanding but would need verifying.